lib/path/default.nix

```
@@ -94,6 +95,47 @@ let
 
 in /* No rec! Add dependencies on this file at the top. */ {
 
+  /* Append a subpath string to an absolute path value, the result is an absolute path value.
+
+    Like `path + ("/" + string)` but safer, because it errors for potentially unexpected results.
+    More specifically, it checks that the first argument is a path value,
+    and that the second argument is a valid subpath string, see `lib.path.subpath.isValid`.
+
+    Type:
+      append :: Path -> String -> Path
+
+    Examples:
+      append /foo "bar/baz"
+      => /foo/bar/baz
+
+      # subpaths don't need to be normalised
+      append /foo "./bar//baz/./"
+      => /foo/bar/baz
+
+      # can append to root directory
+      append /. "foo/bar"
+      => /foo/bar
+
+      # first argument needs to be a Nix path value
```

```suggestion
      # first argument needs to be a Nix path
```

---

This might be especially confusing because the next line errors for `append "/foo" "bar"`, where the first argument does indeed look like a "path". Maybe it should be

```suggestion
      # first argument needs to be a path data type
```

instead